CDCgov/prime-reportstream
# Allow test result in any OBX iteration

This PR adds functionality to accept HL7 messages where the test result is not in the first observation iteration. HL7 does not require a certain order for the observations. But currently RS assumes the first OBX segment will contain the test result. These changes checks the first OBX. If the LOINC code in that OBX-3 field is not present in the LIVD table "Test Performed LOINC Code" column, then the remaining OBX segments are searched. If an OBX is found with a OBX-3 value in the LIVD table then that observation group is moved the top of the grouping. If an OBX is not identified then the message is not changed and moves forward in the process. 

Test Steps:
1. pull and package
2. `./prime test`
3. `./gradlew testIntegration`
4. Send test file
     - `curl -X POST -H "client: neurolab" -H "Content-Type: application/hl7-v2" --data-binary "PATH/TO/test_file.txt" "http://localhost:7071/api/reports"`
     - [test_file.txt](https://github.com/CDCgov/prime-reportstream/files/9974956/test_file.txt)
5. Check output HL7 to ensure that the test result observation group(OBX and NTE) was moved to the beginning of the list and that the OBX Set IDs were re-numbered

## Changes
- add function organizeObservationOrder()
- add function checkLIVDValueExists()

## Checklist

### Testing
- [X] Added `test checkLIVDValueExists` to HL7SerializerTests
- [X] Added `test confirmObservationOrder` to HL7SerializerTests

## Linked Issues
- Fixes #6733
